src/safir/github/models.py

```diff
@@ -0,0 +1,372 @@
+"""Pydantic models for GitHub v3 REST API resources."""
+
+from __future__ import annotations
+
+from base64 import b64decode
+from enum import Enum
+
+from pydantic import BaseModel, Field, HttpUrl
+
+__all__ = [
+    "GitHubRepoOwnerModel",
+    "GitHubUserModel",
+    "GitHubRepositoryModel",
+    "GitHubPullState",
+    "GitHubPullRequestModel",
+    "GitHubBranchCommitModel",
+    "GitHubBranchModel",
+    "GitHubBlobModel",
+    "GitHubCheckSuiteStatus",
+    "GitHubCheckSuiteConclusion",
+    "GitHubCheckSuiteModel",
+    "GitHubCheckRunStatus",
+    "GitHubCheckRunConclusion",
+    "GitHubCheckRunAnnotationLevel",
+    "GitHubCheckSuiteId",
+    "GitHubCheckRunOutput",
+    "GitHubCheckRunPrInfoModel",
+    "GitHubCheckRunModel",
+]
+
+
+class GitHubRepoOwnerModel(BaseModel):
+    """A Pydantic model for the ``owner`` field found in repository objects.
+
+    https://docs.github.com/en/rest/repos/repos#get-a-repository
+    """
+
+    login: str = Field(
+        title="Login name",
+        description=(
+            "Login name of the owner (either a user or an organization)."
+        ),
+        example="lsst-sqre",
+    )
+
+
+class GitHubUserModel(BaseModel):
+    """A Pydantic model for the ``user`` field found in GitHub API resources.
+
+    This contains brief (public) info about a user.
+    """
+
+    login: str = Field(title="Login name", description="GitHub username.")
```

Comment:
It's a shame that the Sphinx Pydantic docs extension seems to mostly throw away the `title` so you have to repeat yourself in `description`, but looking at the rendered docs, this seems like the right approach.

Reply:
Yeah, specifically this is the [autodoc_pydantic](https://autodoc-pydantic.readthedocs.io/en/stable/) extension, where the description is the key documentation field. It's definitely a different treatment from redoc/FastAPI. So the tension will mostly come up if we need to both document a Pydantic model as a Pydantic model and as an OpenAPI schema.